pkg/volume/fc/fc_util.go

```diff
@@ -178,8 +187,18 @@ func (util *FCUtil) AttachDisk(b fcDiskMounter) (string, error) {
 }
 
 func (util *FCUtil) DetachDisk(c fcDiskUnmounter, mntPath string) error {
+	devName, _, err := mount.GetDeviceNameFromMount(c.mounter, mntPath)
```

Comment:
@rootfs 
I think the /dev/sdX and associated symlinks such as /dev/disk/by-id/, /dev/disk/by-path/ are removed by kernel and udev immediately. Therefore I don't think there is race window after the detach operation and next attach operation.

In the future we'll have block volumes support for FC plugin. In this case, we'll need to run "blockdev --flushbufs <device>" operation for flushing I/O before removing SCSI device because I/O might be on block buffer cache of operating system. I suppose this operation might takes several seconds and it depends on how much cache is used to. We should wait this I/O flush and symlink removal using similar way with os-brick. But even this case, I believe race window doesn't happen.

See: 
- https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Storage_Administration_Guide/removing_devices.html

- Also https://github.com/openstack/os-brick/blob/master/os_brick/initiator/linuxscsi.py
wait_for_volumes_removal()